core/platform/datastore/cloud_datastore_services.py

```diff
@@ -0,0 +1,285 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides a seam for datastore services."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import contextlib
+import datetime
+
+import python_utils
+
+from google.cloud import ndb
+
+Model = ndb.Model
+
+DateTimeProperty = ndb.DateTimeProperty
+
+
+Model = ndb.Model
+Key = ndb.Key
+
+BooleanProperty = ndb.BooleanProperty
+DateTimeProperty = ndb.DateTimeProperty
+FloatProperty = ndb.FloatProperty
+IntegerProperty = ndb.IntegerProperty
+JsonProperty = ndb.JsonProperty
+StringProperty = ndb.StringProperty
+TextProperty = ndb.TextProperty
+UserProperty = ndb.UserProperty
+
+
+def get_multi(keys):
+    """Fetches a sequence of keys.
+
+    Args:
+        keys: list(str). The keys to look up.
+
+    Returns:
+        list(ndb.Model). List whose items are either a Model instance or None if
+        the key wasn't found.
+    """
+    return ndb.get_multi(keys)
+
+
+def put_multi(models, update_last_updated_time=True):
+    """Stores a sequence of Model instances.
+
+    Args:
+        models: datastore_services.Model. A sequence of Model instances.
+        update_last_updated_time: bool. Whether to update the last_updated field
+            of the entities.
+
+    Returns:
+        list(str). A list with the stored keys.
+    """
+    del update_last_updated_time # No longer supported.
+    return ndb.put_multi(models)
+
+
+def put_multi_async(models, update_last_updated_time=True):
+    """Stores a sequence of Model instances asynchronously.
+
+    Args:
+        models: datastore_services.Model. A sequence of Model instances.
+        update_last_updated_time: bool. Whether to update the last_updated field
+            of the entities.
+
+    Returns:
+        list(future). A list of futures.
+    """
+    del update_last_updated_time # No longer supported.
+    return ndb.put_multi_async(models)
+
+
+def delete_multi(keys):
+    """Deletes a sequence of keys.
+
+    Args:
+        keys: list(str). A sequence of keys.
+
+    Returns:
+        list(None). A list of Nones, one per deleted model.
```

Comment:
What is the point of returning a list of Nones?

Is this what was done before?

Reply:
Yeah, this is what the actual API returns. I suppose it's useful for matching the pattern of all other model APIs (returning a sequence matching the size of input).